src/components/group-common.tsx

```diff
@@ -0,0 +1,144 @@
+import { makeMenuIcon, MenuTrigger } from 'src/components/PopupTrigger';
+import { ariaSort, HeaderRenderer } from 'src/components/table';
+import { BillingRole } from 'src/libs/ajax/Billing';
+import { GroupRole } from 'src/libs/ajax/Groups';
+import { cardList as cardListStyles } from 'src/libs/style';
+
+interface AdminNotifierCheckboxProps {
+  checked: boolean;
+  onChange: (v: boolean) => void;
+}
+export const AdminNotifierCheckbox = (props: AdminNotifierCheckboxProps) => (
+  <div style={{ marginTop: '0.5rem', display: 'flex', alignItems: 'center' }}>
+    <LabeledCheckbox style={{ marginRight: '0.25rem' }} checked={props.checked} onChange={props.onChange}>
+      Allow anyone to request access
+    </LabeledCheckbox>
+    <InfoBox style={{ marginLeft: '0.3rem' }}>
+      Any user will be able to request to become a member of this group. This will send an email to the group admins.
+    </InfoBox>
+  </div>
+);
+
+interface NewUserCardProps {
+  onClick: () => void;
+}
+export const NewUserCard = (props: NewUserCardProps) => (
+  <ButtonPrimary style={{ textTransform: 'none' }} onClick={props.onClick}>
+    <Icon icon='plus' size={14} />,<div style={{ marginLeft: '0.5rem' }}>Add User</div>
+  </ButtonPrimary>
+);
+
+interface UserMenuContentProps {
+  onEdit: () => void;
+  onDelete: () => void;
+}
+
+const UserMenuContent = (props: UserMenuContentProps) => (
+  <>
+    <MenuButton onClick={props.onEdit}>
+      {makeMenuIcon('edit')}
+      Edit Role
+    </MenuButton>
+    ,
+    <MenuButton onClick={props.onDelete}>
+      {makeMenuIcon('trash')}
+      Remove User
+    </MenuButton>
+  </>
+);
+
+const menuCardSize = 20;
+
+export interface Sort {
+  field: string;
+  direction: 'asc' | 'desc';
+}
+
+interface MemberCardHeadersProps {
+  sort: Sort;
+  onSort: (v: Sort) => void;
+}
+
+export const MemberCardHeaders = (props: MemberCardHeadersProps): ReactNode => (
+  <div role='row' style={{ display: 'flex', justifyContent: 'space-between', marginTop: '1.5rem', padding: '0 1rem' }}>
+    <div role='columnheader' aria-sort={ariaSort(props.sort, 'email')} style={{ flex: 1 }}>
+      <HeaderRenderer sort={props.sort} onSort={props.onSort} name='email' />
+    </div>
+    <div role='columnheader' aria-sort={ariaSort(props.sort, 'roles')} style={{ flex: 1 }}>
+      <HeaderRenderer sort={props.sort} onSort={props.onSort} name='roles' />
+    </div>
+    <div
+      role='columnheader'
+      // Width is the same as the menu icon.
+      style={{ width: menuCardSize }}
+    >
+      <div className='sr-only'>Actions</div>
+    </div>
+  </div>
+);
+
+export type UserRole = BillingRole | GroupRole;
+
+export interface User {
+  email: string;
+  roles: UserRole[];
+}
+
+interface MemberCardProps {
+  member: User;
+  adminCanEdit: boolean;
+  onEdit: () => void;
+  onDelete: () => void;
+  adminLabel: string;
+  userLabel: string;
+  isOwner: boolean;
+}
+
+export const MemberCard = (props: MemberCardProps): ReactNode => {
```

Comment:
As with `GroupCard`, this component was also memoized. 
This is a very simple component, and I can't see it making sense to memoize it unless you had thousands of members and frequent sorts. Even then, it likely wouldn't work, sense the components would need to be rendered in a different place in the page.

Just in case, I did test this without memoizing the component, on a group with about 100 members, resorting by columns as fast as I could click, with no issues.

Reply:
I just realized, the `onSort` being passed in the props will change when the parent re-renders, so the memo will never have any effect. 
We'd need to store the value in the parent with a `useMemo` hook.